drivers/dma/dma_stm32.c

```diff
@@ -134,15 +134,32 @@ static void dma_stm32_irq_handler(const struct device *dev, uint32_t id)
 
 static void dma_stm32_shared_irq_handler(const struct device *dev)
 {
-	const struct dma_stm32_config *cfg = dev->config;
-	DMA_TypeDef *dma = (DMA_TypeDef *)(cfg->base);
+	ARG_UNUSED(dev);
 	uint32_t id = 0;
 
-	for (id = 0; id < cfg->max_streams; ++id) {
-		if (stm32_dma_is_irq_active(dma, id)) {
-			dma_stm32_irq_handler(dev, id);
+#if DT_NODE_HAS_STATUS(DT_DRV_INST(0), okay)
+	static const struct device *dev0 = DEVICE_DT_INST_GET(0);
+	const struct dma_stm32_config *cfg0 = dev0->config;
+	DMA_TypeDef *dma0 = (DMA_TypeDef *)(cfg0->base);
+
+	for (id = 0; id < cfg0->max_streams; ++id) {
+		if (stm32_dma_is_irq_active(dma0, id)) {
+			dma_stm32_irq_handler(dev0, id);
+		}
+	}
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_DRV_INST(1), okay)
+	static const struct device *dev1 = DEVICE_DT_INST_GET(1);
+	const struct dma_stm32_config *cfg1 = dev1->config;
+	DMA_TypeDef *dma1 = (DMA_TypeDef *)(cfg1->base);
+
+	for (id = 0; id < cfg1->max_streams; ++id) {
+		if (stm32_dma_is_irq_active(dma1, id)) {
+			dma_stm32_irq_handler(dev1, id);
 		}
 	}
+#endif
```

Comment:
Instance 0 is always there so the directive checking it is enabled is superfluous.
So either:
- remove this initial instance 0 check
- take full advantage of instances to factorize a bit this part (using pre-processor)

Reply:
using DT_INST_FOREACH_STATUS_OKAY instead.